syntax = "proto3";
package boiler;

service BoilerPhysics {
    rpc GetStatus (Empty) returns (BoilerStatus);
    rpc SetControls (ControlInput) returns (BoilerStatus);
}

message Empty{}

message BoilerStatus {
    double timestamp = 1;
    double furnace_temp = 2;
    double steam_pressure = 3;
    double drum_level = 4;
    double steam_flow = 5;
}

message ControlInput{
    double fuel_valve = 1;
    double feedwater_valve = 2;
    double steam_valve = 3;
}